src/main/scala/com/databricks/spark/sql/perf/mllib/fpm/FPGrowth.scala

```diff
@@ -0,0 +1,36 @@
+package com.databricks.spark.sql.perf.mllib.fpm
+
+import org.apache.spark.ml
+import org.apache.spark.ml.PipelineStage
+import com.databricks.spark.sql.perf.mllib.OptionImplicits._
+import com.databricks.spark.sql.perf.mllib._
+import com.databricks.spark.sql.perf.mllib.data.DataGenerator
+import org.apache.spark.sql.DataFrame
+
+/** Object containing methods used in performance tests for FPGrowth */
+object FPGrowth extends BenchmarkAlgorithm with TestFromTraining {
+
+  def trainingDataSet(ctx: MLBenchContext): DataFrame = {
+    import ctx.params._
+    val rng = ctx.newGenerator()
+
+    // These two param will move to MLParam later.
+    val numItems = 1000
+    val averageSizeOfItemSet = 4
```

Comment:
This two parameters should move into `MLParam`, but `MLParam` need some refactor because now it cannot hold more than 22 values. I do not want do the change again so pending https://github.com/databricks/spark-sql-perf/pull/112 merge.